test/lib/commands/copy.js

```diff
@@ -0,0 +1,221 @@
+const t = require('tap')
+const { load } = require('../../fixtures/mock-npm')
+const path = require('path')
+const fs = require('fs')
+
+const cwd = process.cwd()
+t.afterEach(t => process.chdir(cwd))
+
+t.test('should copy module files to destination', async t => {
+  const { npm, outputs, logs } = await load(t, {
+    prefixDir: {
+      'package.json': JSON.stringify({
+        name: 'test-package',
+        version: '1.0.0',
+        files: ['lib'],
+      }),
+      'README.md': 'file',
+      lib: {
+        'index.js': '// empty',
+      },
+      src: {
+        'index.js': '// empty',
+      },
+    },
+  })
+  process.chdir(npm.prefix)
+  await npm.exec('copy', ['build'])
+  t.strictSame(outputs, [])
+  t.strictSame(logs.notice, [])
+
+  // lib is included in files, while package.json and README.md are always included.
+  assertExists(path.join('build', 'package.json'))
+  assertExists(path.join('build', 'README.md'))
+  assertExists(path.join('build', 'lib', 'index.js'))
+
+  // src should not be copied because it's excluded by files.
+  assertMissing(path.join('build', 'src'))
+})
+
+t.test('should copy dependencies', async t => {
+  const { npm, outputs, logs } = await load(t, {
+    prefixDir: {
+      'package.json': JSON.stringify({
+        name: 'test-package',
+        version: '1.0.0',
+        dependencies: {
+          foo: '^1.0.0',
+          bar: '^1.0.0',
+        },
+      }),
+      node_modules: {
+        foo: { },
+        bar: { },
+        baz: { },
+      },
+    },
+  })
+  process.chdir(npm.prefix)
+  await npm.exec('copy', ['build'])
+  t.strictSame(outputs, [])
+  t.strictSame(logs.notice, [])
+
+  assertExists(path.join('build', 'node_modules', 'foo'))
+  assertExists(path.join('build', 'node_modules', 'bar'))
+  // baz is missing because it is an extraneous dep.
+  assertMissing(path.join('build', 'node_modules', 'baz'))
+})
+
+t.test('should not copy bundled dependencies if they are omitted', async t => {
+  const { npm, outputs, logs } = await load(t, {
+    prefixDir: {
+      'package.json': JSON.stringify({
+        name: 'test-package',
+        version: '1.0.0',
+        bundledDependencies: true,
+        optionalDependencies: {
+          foo: '^1.0.0',
+        },
+      }),
+      node_modules: {
+        foo: {
+          'package.json': JSON.stringify({
+            name: 'foo',
+            version: '1.0.0',
+          }),
+        },
+      },
+    },
+  })
+  process.chdir(npm.prefix)
+  npm.config.set('omit', ['optional'])
```

Comment:
Is the name of the test correct? We omit not bundled, but optional dependencies here.